src/components/user/EditProfile.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useMemo, useState } from 'react'
+import Image from 'next/image'
+import imageIcon from '@icons/image_icon.svg'
+import deleteIcon from '@icons/close_icon.svg'
+import CustomButton from '../@shared/button/CustomButton'
+import { Spinner } from 'flowbite-react'
+
+const INPUT_MAX_LENGTH = 10
+const TEXTAREA_MAX_LENGTH = 300
+
+type ProfileContentsTypes = {
+  image: string | File | null
+  nickname: string
+  description: string
+}
+
+interface EditProfileProps extends ProfileContentsTypes {
+  onEdit: (params: ProfileContentsTypes) => void
+  isPending: boolean
+}
+
+export default function EditProfile({ image, nickname, description, onEdit, isPending }: EditProfileProps) {
+  const [formValues, setFormValues] = useState({
+    image,
+    nickname,
+    description,
+  })
+  const [previewImage, setPreviewImage] = useState<string | File | null>(image ?? null)
+  const [inputCount, setInputCount] = useState<number>(description.length)
+
+  const isFormComplete = useMemo(() => {
+    const { image, ...restValues } = formValues
+    const isAllInputFilled = Object.values(restValues).every((inputValue) => inputValue !== '')
+    return isAllInputFilled
+  }, [formValues])
+
+  const handleFileInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!e.target.files) return
+
+    const selectedImageFile = e.target.files[0]
+
+    if (previewImage && typeof previewImage === 'string') {
+      URL.revokeObjectURL(previewImage)
+    }
+
+    setPreviewImage(URL.createObjectURL(selectedImageFile))
```

Comment:
```suggestion
    const nextImage = URL.createObjectURL(selectedImageFile)
    setPreviewImage(nextImage)
```
가독성, 디버깅 용이성 등등을 위해 이렇게 쓰는 것이 좋을 것 같습니다.